virtcontainers/api.go

```diff
@@ -74,6 +74,12 @@ func createSandboxFromConfig(ctx context.Context, sandboxConfig SandboxConfig, f
 		return nil, err
 	}
 
+	defer func() {
+		if err != nil {
+			rollbackFailingSandboxCreation(ctx, s)
```

Comment:
There's a problem with calling this function here. It contains a call deleting the sandbox resources, but they are created by `storeSandbox()`, which means they are not created yet at this point in the code.

Reply:
Yes you're right. But what I'm trying to do is define a global rollback function and is put to `defer` at the very beginning of function. 

Or what you prefer is smaller rollback operations and adds after each operations?

Reply:
Because you're calling the defer right after `createSandbox()` succeeded, then we should expect to rollback only things that have been set up by `createSandbox()`.
So I guess to answer your question, several smaller rollbacks would be more appropriate.

Reply:
Updated according to your comments, please help to check again, thanks :smiley: